### PR TITLE
Delegated Manager System audit fixes and upgrades [SIM-55] [SIM-156]

### PR DESCRIPTION
**Pull request by pblivin0x**
Audit Fixes

[5.3] Potential grief through arbitrary contract call
- Add `controller` to DelegatedManagerFactory. 
    -  In `createManager()`, require that the incoming `_setToken` is valid on the `controller`
    - In `initialize()`, require that the incoming `_initializeTargets` does not include a `controller` tracked `setToken`

[5.4.1] Design Comments
- On DelegatedManagerFactory, make `setTokenFactory` immutable
- On BaseGlobalExtension, make `managerCore` immutable
- On DelegatedManager, add address zero check to `setMethodologist()`

Upgrades
- On DelegatedManager, add mutual upgrade between the `owner` and `methodologist` to `updateOwnerFeeSplit()`

### CODE REVIEW

File: contracts/lib/MutualUpgradeV2.sol [hunk 1]
```diff
@@ -0,0 +1,79 @@
+/*
```
Comment (bweick) [inline contracts/lib/MutualUpgradeV2.sol:1]: Commenting the diff **[here](https://gist.github.com/bweick/79c26c48c6c6e7cf7db8e3cb3e7e5bf2)**

File: test/manager/delegatedManager.spec.ts [hunk 2]
```diff
@@ -776,17 +781,33 @@ describe("DelegatedManager", () => {
...
+        const txHash = await subject(subjectOwnerCaller);
+
+        const expectedHash = solidityKeccak256(
+          ["bytes", "address"],
+          [txHash.data, subjectOwnerCaller.address]
+        );
+
+        const isLogged = await delegatedManager.mutualUpgrades(expectedHash);
+
+        expect(isLogged).to.be.true;
```
Comment (bweick) [inline test/manager/delegatedManager.spec.ts:800]: Let's also just check here to make sure that the state wasn't updated as well

**Review by cgewecke — COMMENTED** [review]
The DelegatedManager isn't auto-verifying on Etherscan. 

I think this is because the `setToken` and `factory` variables are immutable and those addresses get embedded in the bytecode. 

When a new DelegatedManager is created it will always vary slightly from all the others - Etherscan doesn't get enough context from a factory deployment to verify that these are all variants of the same contract.  

TLDR; suggest removing the `immutable` keyword from both of those state variables.

File: contracts/factories/DelegatedManagerFactory.sol [hunk 3]
```diff
@@ -234,23 +241,27 @@ contract DelegatedManagerFactory {
         require(msg.sender == initializeState[_setToken].deployer, "Only deployer can initialize manager");
         _initializeTargets.validatePairsWithArray(_initializeBytecode);
 
-        IDelegatedManager manager = initializeState[_setToken].manager;
```
Comment (pblivin0x) [inline contracts/factories/DelegatedManagerFactory.sol:237]: moving the new require statement above the owner fee updates

Comment (bweick) [reply contracts/factories/DelegatedManagerFactory.sol:237]: +1

File: contracts/factories/DelegatedManagerFactory.sol [hunk 4]
```diff
@@ -91,15 +95,18 @@ contract DelegatedManagerFactory {
     /**
      * @dev Sets managerCore and setTokenFactory address.
      * @param _managerCore                      Address of ManagerCore protocol contract
+     * @param _controller                       Address of Controller protocol contract
      * @param _setTokenFactory                  Address of SetTokenFactory protocol contract
      */
     constructor(
         IManagerCore _managerCore,
+        IController _controller,
```
Comment (bweick) [inline contracts/factories/DelegatedManagerFactory.sol:103]: Just noting this will cause some changes in the deploy scripts

**Review by bweick — APPROVED** [review]
